.infrastructure/identity/main.tf

```diff
@@ -122,3 +122,48 @@ resource "aws_iam_role_policy_attachment" "githubiac" {
   role       = aws_iam_role.githubiac.name
   policy_arn = data.aws_iam_policy.admin_access.arn
 }
+
```

Comment:
The intention is to keep all the resources needed by Terraform itself (Terraform state + iam to apply) in the "identity" module. IMHO the role used to deploy the application should not be there. If you agree we can move it  under '.infrastructure/iam.tf'.